module/VuFind/config/module.config.php

```diff
@@ -520,7 +520,8 @@
             'VuFind\Translator' => 'Laminas\Mvc\I18n\Translator',
             'VuFind\WorldCatUtils' => 'VuFind\Connection\WorldCatUtils',
             'VuFind\YamlReader' => 'VuFind\Config\YamlReader',
-            'Laminas\Validator\Csrf' => 'VuFind\Validator\Csrf',
+            'Laminas\Validator\Csrf' => 'VuFind\Validator\CsrfInterface',
```

Comment:
Interface is what the service must provide, but since there could be several classes providing the interface, ot seems a bit backwards to me to handle services by the interface. I'd rather map `'Laminas\Validator\Csrf' => 'VuFind\Validator\Csrf'` and `'VuFind\Validator\Csrf' => 'VuFind\Validator\SessionCsrf'`.

Reply:
Should I use string `VuFind\Validator\Csrf` or non-existing class `\VuFind\Validator\Csrf:class`  in factories?

Reply:
Good question, I'm not sure about that. Maybe @demiankatz has a further idea. Since the plain Csrf class doesn't exist it might be better to use `\Laminas\Validator\Csrf::class => \VuFind\Validator\SessionCsrf::class` and `'\VuFind\Validator\Csrf' => \VuFind\Validator\SessionCsrf::class` instead.

Reply:
I think the only place we should use the string 'VuFind\Validator\Csrf' is in the module configuration to define an alias for backward compatibility. In any code that requests a service, we should pull either \VuFind\Validator\CsrfInterface::class (if we need a service that satisfies this interface, but we don't care which one), or \VuFind\Validator\SessionCsrf::class (if there is a particular reason why we would want specifically this implementation). I don't think it's a problem that there might be multiple implementations of the interface in the code; the idea of requesting an interface from the service manager is to find out which implementation of the class is configured to be preferred. I suspect in the future if we add multiple CSRF implementations, we'll change our factories anyway to use some kind of plugin manager and new configuration.

I realize this requires @xrosecky to undo some work from the most recent commit. I apologize for not getting this response written a little faster!

Reply:
Ok. Sorry for making a mess out of this, @xrosecky.

Reply:
@demiankatz OK, I added VuFind\Validator\Csrf alias.